test/lib/types/application/ApplicationFormatter.js

```diff
@@ -96,6 +96,16 @@ test("validate: empty resources", async (t) => {
 	t.deepEqual(myProject.resources.configuration.paths.webapp, "webapp", "default webapp directory is set");
 });
 
+test("validate: test invalid encoding", async (t) => {
+	const myProject = clone(applicationBTree);
+	myProject.resources.configuration.propertiesFileSourceEncoding = "test";
+	const applicationFormatter = new ApplicationFormatter({project: myProject});
+
+	const error = await t.throwsAsync(applicationFormatter.validate(myProject));
+	t.is(error.message, `Invalid properties file encoding specified for project application.b: encoding provided: test. Must be either "ISO-8859-1" or "UTF-8".`,
+		"Missing source directory caused error");
+});
+
```

Comment:
The only thing I'm missing here is a test for the defaulting. An explicit test that a project without a `propertiesFileSourceEncoding` configuration, is being defaulted to having a `propertiesFileSourceEncoding: "ISO-8859-1"` configuration.

For the LibraryFormatter this assert has been added to the test `validate: test directory does not exist`. But a separate test would be even better.

Reply:
done

Reply:
fixed with: 30ada8b